src/pymedphys/dicom/_level2/dicom_dose.py

```diff
@@ -451,6 +455,22 @@ def load_dicom_data(ds, depth_adjust):
     return inplane, crossplane, depth, dose
 
 
+def interpolating_profile_extract(ds, depth_adjust, inplane_ref, crossplane_ref, depth_ref):
```

Comment:
Slightly odd function name. Ordinarily, the verb should be the first word in the function name. E.g. `extract_interpolated_profile`.
Or is "extract" a noun in this context? Even then, `interpolate` would be preferred over `interpolating`.

Reply:
I changed this to `arbitrary_profile_extract_from_ds`, what do you think? I'm not sold...

Reply:
Thought about it a moment longer, `extract_arbitrary_profile_from_ds` is better.

Reply:
Or `arbitrary profile from dicom dataset`? Or even `arbitrary profile from dicom dose`? No need to stick to `ds` for function names.

Reply:
Yes, `arbitrary_profile_from_dicom_dose`. I've been getting to the point that I have just wanted to implement our full DICOM object stack that we were referring to. It would make it possible to create tests for these kinds of functions in quite a nice neat way...

Reply:
Hahaha, I hear you :)

Reply:
:)